### PR TITLE
Mohammad darwesh

### PR DESCRIPTION
**Pull request by mohammadnaefaldarwesh**

### CODE REVIEW

File: Week2/MohammadDarwesh/mohammaddarwesh.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
+// for the first task i will do it with many ways just for practising.
```
Comment (paulbremer) [inline Week2/MohammadDarwesh/mohammaddarwesh.js:1]: Hi Mohammad, the homework also contains the code from week2, next time you should make a new branch from `master` so only the code for week3 is in the pull request.

File: Week3/Mohammad-Darwesh-homework-JS1-week3/mohammaddarwesh.js [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+let favoriteAnimals = ["blowfish", "capricorn", "giraffe"];
+console.log(favoriteAnimals);
+favoriteAnimals.push("turtle");
+console.log(favoriteAnimals);
+favoriteAnimals.splice (1, 0, "meerkat");
+console.log("I think the new value of the array is [\"blowfish\", \"meerkat\", \"capricorn\", \"giraffe\", \"turtle\"]");
+console.log(favoriteAnimals);
+console.log("The array has a length of",favoriteAnimals.length);
+// or :
+const l = favoriteAnimals.length;
```
Comment (paulbremer) [inline Week3/Mohammad-Darwesh-homework-JS1-week3/mohammaddarwesh.js:22]: Be careful with the names of variables, you know what `l` means right now but the next developer probably doesn't know. So it's better to give variable the correct names, `animalsLength`  might be better here.

File: Week3/Mohammad-Darwesh-homework-JS1-week3/mohammaddarwesh.js [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+};
+function myInformation (me){
+    return me;
+}
+console.log(myInformation(me));
+
+// task 4:
+function vehicleType (color, x){
+    if (x === 1){
+        return  `a ${color} car`;
```
Comment (paulbremer) [inline Week3/Mohammad-Darwesh-homework-JS1-week3/mohammaddarwesh.js:73]: A nice use of backticks ` `` `!

File: Week3/Mohammad-Darwesh-homework-JS1-week3/mohammaddarwesh.js [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+console.log(z === y);
+function same(a,b,c){
+    let same = a.length === b.length && a.length === c.length && b.length === c.length,
+        result,
+        len;
+    if(!same) { result = false }
+    else {
+        result = true;
+        len = a.length;
+        for(let i=0;result !== false && i < len; i++) {
```
Comment (paulbremer) [inline Week3/Mohammad-Darwesh-homework-JS1-week3/mohammaddarwesh.js:213]: You assigned `true` to result on line 211, so no need to check if its true here.

**Review by paulbremer — APPROVED** [review]
Good job on the homework! Next time try to use clear variable names, if the task refers to `code` try to use the same `code` in your function instead of `x`. If you are working with other developers these things are really important. Let me know if you have any questions!

File: Week3/Mohammad-Darwesh-homework-JS1-week3/mohammaddarwesh.js (hunk 2 above)
Comment (mohammadnaefaldarwesh) [reply Week3/Mohammad-Darwesh-homework-JS1-week3/mohammaddarwesh.js:22]: Hello Paul,
Thank you for your feedback .
I will use your useful advice on next homework.

**Pull request closed without merging**